### PR TITLE
test/SSAD-0008: write unit test for  StreetCode/RelatedFigure GetBySt…

### PR DESCRIPTION
**Pull request by IrynaNakoneshniuk**

### CODE REVIEW

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Streetcode/RelatedFigure/GetByStreetcodeId/GetRelatedFiguresByStreetcodeIdHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+using Moq;
+using Xunit;
+using AutoMapper;
+using System.Reflection;
+using System.Linq.Expressions;
+using Microsoft.EntityFrameworkCore.Query;
+using AutoMapper.Internal;
+using System;
+using System.Linq;
+using global::Streetcode.DAL.Repositories.Interfaces.Base;
```
Comment (vladtymo) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Streetcode/RelatedFigure/GetByStreetcodeId/GetRelatedFiguresByStreetcodeIdHandlerTests.cs:12]: Remove these global namespaces after refactoring namespace names

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Streetcode/RelatedFigure/GetByStreetcodeId/GetRelatedFiguresByStreetcodeIdHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    public GetRelatedFiguresByStreetcodeIdHandlerTests()
+    {
+        this.mapperMock = new Mock<IMapper>();
+        this.repositoryWrapperMock = new Mock<IRepositoryWrapper>();
+        this.loggerMock = new Mock<ILoggerService>();
+        this.handler = new GetRelatedFiguresByStreetcodeIdHandler(this.mapperMock.Object, this.repositoryWrapperMock.Object, this.loggerMock.Object);
+    }
+
+    [Fact]
+    public void GetRelatedFigureIdsByStreetcodeId_ShouldReturnNull_WhenArgumentNullExceptionIsThrown()
```
Comment (vladtymo) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Streetcode/RelatedFigure/GetByStreetcodeId/GetRelatedFiguresByStreetcodeIdHandlerTests.cs:36]: Remove private method tests

**Review by vladtymo — CHANGES_REQUESTED** [review]

**Review by BohdanLytvynov — CHANGES_REQUESTED** [review]
Do not test private methods

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Streetcode/RelatedFigure/GetByStreetcodeId/GetRelatedFiguresByStreetcodeIdHandlerTests.cs (hunk 1 above)
Comment (IrynaNakoneshniuk) [reply Streetcode/Streetcode.XUnitTest/MediatRTests/Streetcode/RelatedFigure/GetByStreetcodeId/GetRelatedFiguresByStreetcodeIdHandlerTests.cs:12]: Fixed

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Streetcode/RelatedFigure/GetByStreetcodeId/GetRelatedFiguresByStreetcodeIdHandlerTests.cs (hunk 2 above)
Comment (IrynaNakoneshniuk) [reply Streetcode/Streetcode.XUnitTest/MediatRTests/Streetcode/RelatedFigure/GetByStreetcodeId/GetRelatedFiguresByStreetcodeIdHandlerTests.cs:36]: Fixed

**Review by BohdanLytvynov — CHANGES_REQUESTED** [review]
1 Assert per 1 testing function.

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/RelatedFigureTests/GetByStreetcodeId/GetRelatedFiguresByStreetcodeIdHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+        this.repositoryWrapperMock.Setup(r => r.StreetcodeRepository
+        .GetAllAsync(It.IsAny<Expression<Func<StreetcodeContent, bool>>>(), It.IsAny<Func<IQueryable<StreetcodeContent>, IIncludableQueryable<StreetcodeContent, object>>>()))
+                              .ReturnsAsync(relatedFigures.AsQueryable());
+
+        var result = await this.handler.Handle(new GetRelatedFigureByStreetcodeIdQuery(streetcodeId), CancellationToken.None);
+
+        Assert.True(result.IsSuccess);
+    }
+
```
Comment (BohdanLytvynov) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/RelatedFigureTests/GetByStreetcodeId/GetRelatedFiguresByStreetcodeIdHandlerTests.cs:62]: This last two methods are the duplications. They provide the same logic.

Comment (IrynaNakoneshniuk) [reply Streetcode/Streetcode.XUnitTest/MediatRTests/StreetcodeTests/RelatedFigureTests/GetByStreetcodeId/GetRelatedFiguresByStreetcodeIdHandlerTests.cs:62]: Fixed

**Review by BohdanLytvynov — COMMENTED** [review]
Delete equal functions

**Review by BohdanLytvynov — APPROVED** [review]

**Review by metrondir — APPROVED** [review]
LGTM!

**Review by BohdanLytvynov — APPROVED** [review]

**Pull request closed without merging**